integration-tests/tests/src/tests/objects.ts

```diff
@@ -24,138 +24,147 @@ import {
-      });
+        const john = realm.write(() => {
+          return realm.create<IPersonWithId>(PersonSchemaWithId.name, {
+            _id,
+            name: "John Doe",
+            age: 42,
+          });
+        });
 
-      expect(john._id.equals(_id)).equals(true);
-      expect(john.name).equals("Mr. John Doe");
-      expect(john.age).equals(43);
+        expect(john._id.equals(_id)).equals(true);
+        expect(john.name).equals("John Doe");
+        expect(john.age).equals(42);
 
-      expect(() =>
         realm.write(() => {
-          realm.create<IPersonWithId>(
-            PersonSchemaWithId.name,
-            { _id, name: "John Doe", age: 42 },
-            Realm.UpdateMode.Never,
-          );
-        }),
-      ).throws(
-        `Attempting to create an object of type '${PersonSchemaWithId.name}' with an existing primary key value '${_id}'.`,
-      );
+          realm.create<IPersonWithId>(PersonSchemaWithId.name, { _id, age: 43 }, Realm.UpdateMode.All);
+        });
 
-      // Expect only one instance of 'PersonSchemaWithId' in db after all updates
-      const persons = realm.objects(PersonSchemaWithId.name);
-      expect(persons.length).equals(1);
-    });
+        expect(john._id.equals(_id)).equals(true);
+        expect(john.name).equals("John Doe");
+        expect(john.age).equals(43);
 
-    it("can return a value on write", () => {
-      const realm = new Realm({ schema: [PersonSchema] });
+        const update: Partial<IPersonWithId> = {
+          _id,
+          name: "Mr. John Doe",
+        };
 
-      const john = realm.write(() => {
-        return realm.create<IPerson>(PersonSchema.name, {
-          name: "John Doe",
-          age: 42,
+        realm.write(() => {
+          realm.create<IPersonWithId>(PersonSchemaWithId.name, update, Realm.UpdateMode.Modified);
         });
-      });
 
-      // Expect John to be the one and only result
-      const persons = realm.objects(PersonSchema.name);
-      expect(persons.length).equals(1);
-      const [firstPerson] = persons;
-      expect(firstPerson).deep.equals(john);
+        expect(john._id.equals(_id)).equals(true);
+        expect(john.name).equals("Mr. John Doe");
+        expect(john.age).equals(43);
+
+        expect(() =>
+          realm.write(() => {
+            realm.create<IPersonWithId>(
+              PersonSchemaWithId.name,
+              { _id, name: "John Doe", age: 42 },
+              Realm.UpdateMode.Never,
+            );
+          }),
+        ).throws(
+          `Attempting to create an object of type '${PersonSchemaWithId.name}' with an existing primary key value '${_id}'.`,
+        );
+
+        // Expect only one instance of 'PersonSchemaWithId' in db after all updates
+        const persons = realm.objects(PersonSchemaWithId.name);
+        expect(persons.length).equals(1);
+      });
     });
   });
 
   describe("Class Model", () => {
+    beforeEach(() => {
+      Realm.clearTestState();
```

Comment:
I guess this could also be abstracted into a beforeHook?

Reply:
I though the same, but choose not to, since I want the test to fail and not the before-hook, in case the class based models feature breaks.